### ISSUE TITLE
Missing feature: TypeScript association getters & setters

### ISSUE THREAD

**Issue by KjellVermeersch** [sequelize/sequelize-auto#452]
SequelizeAuto now supports TypeScript, but there are still missing features which make the use of Sequelize the most usefull. 
One of the important missing features are the auto generated association getters and setters. By this I mean the getters and setters that are available when using hasMany and belongsToMany assiciations.

In [this document](https://github.com/sequelize/sequelize/blob/9c446f9c96dcf00147a296b09263a73b9b26b03e/docs/manual/other-topics/typescript.md) of Sequelize, they tell you how to implement them into the TypeScript code. This needs to be done "manually" because of the way that TypeScript works. 

```TS
// Since TS cannot determine model association at compile time
// we have to declare them here purely virtually
// these will not exist until `Model.init` was called.
public getProjects!: HasManyGetAssociationsMixin<Project>; // Note the null assertions!
public addProject!: HasManyAddAssociationMixin<Project, number>;
public hasProject!: HasManyHasAssociationMixin<Project, number>;
public countProjects!: HasManyCountAssociationsMixin;
public createProject!: HasManyCreateAssociationMixin<Project>;
```

Maybe this example is missing a set function for many-to-many relationships, but i am not sure.
Is it possible to add these for the TypeScript generated models?

**Comment by steveschmitt** [comment]
Yes it is possible.  It would require some refactoring of the code to put the foreign-key processing before the class generation.

**Comment by steveschmitt** [comment]
Fixed in 0.7.5
